packages/runtime-core/src/hmr.ts

```diff
@@ -27,14 +27,7 @@ export interface HMRRuntime {
 // Note: for a component to be eligible for HMR it also needs the __hmrId option
 // to be set so that its instances can be registered / removed.
 if (__DEV__) {
-  const globalObject: any =
-    typeof global !== 'undefined'
-      ? global
-      : typeof self !== 'undefined'
-        ? self
-        : typeof window !== 'undefined'
-          ? window
-          : {}
+  const globalObject = getGlobalThis()
 
   globalObject.__VUE_HMR_RUNTIME__ = {
```

Comment:
```suggestion
  getGlobalThis().__VUE_HMR_RUNTIME__ = {
```

Could just inline this.

Reply:
nice suggestion~
done!